### PR TITLE
User: keep logged in when reloads

### PR DESCRIPTION
**Pull request by y-c-wang**
# Description

Restore login status after refresh

# Changes

- `loadCurrentUser` to retrieve the current user after `onAuthStateChanged` finished
- Add loading skeleton for user avatar in UserFab

# Notes

image

# Checklist

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] Any changes to strings have been published to our translation tool
- [x] I conducted basic QA to assure all features are working
- [x] I requested code review from other team members

# Resolved Issues

#71

### CODE REVIEW

File: src/utils/firebase/auth.ts [hunk 1]
```diff
@@ -13,6 +15,23 @@ import { firebaseEmulatorConfig } from "../../constants/firebase";
...
+      if (currentUser !== undefined) {
+        resolve(currentUser);
+      } else {
+        setTimeout(checkVariable, 100);
+      }
+    };
+
+    checkVariable();
+  });
+
```
Comment (linyc0817) [inline src/utils/firebase/auth.ts:34]:
```suggestion
export const loadCurrentUser = (): Promise<User | null> =>
  new Promise((resolve) => {
    const unsubscribe = onAuthStateChanged(firebaseAuth, (user) => {
      resolve(user);
      unsubscribe();
    });
  });
```

Will this be better? It seems like your origin method will keep listening every 100ms. Which might let efficacy go down.

**Review by linyc0817 — CHANGES_REQUESTED** [review]
Excellent! But i wonder why `setPersistence` not working for us?

File: src/utils/firebase/auth.ts (hunk 1 above)
Comment (y-c-wang) [reply src/utils/firebase/auth.ts]: Updated. Thanks

**Comment by y-c-wang** [comment]
> Excellent! But i wonder why `setPersistence` not working for us?

the default persistence is what we needed.
Instead, we need to listen to the change of auth state and set it up.

**Pull request merged**